### PR TITLE
Update RTD configuration

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -185,12 +186,12 @@
 # -- Options for LaTeX output --------------------------------------------------
```
Comment (willkg) [inline docs/conf.py:186]: If you do a `sphinx-quickstart` in a new project, it produces a very very very small `conf.py` file that doesn't have a lot of these options. I did a pass on some of my libraries to reduce the `conf.py` file. Maybe we do a pass on reducing the `conf.py` file in this PR?

For example, here's the Markus one: https://github.com/willkg/markus/blob/main/docs/conf.py

**Review by willkg — APPROVED** [review]
This looks fine to me, but I'd probably take a pass at reducing `conf.py`, too.

File: docs/conf.py [hunk 2]
```diff
@@ -24,251 +17,32 @@
 
 os.environ["DJANGO_SETTINGS_MODULE"] = "tests.settings"
 
-import django
-import mozilla_django_oidc
+import django  # noqa
 
+import mozilla_django_oidc  # noqa
```
Comment (willkg) [inline docs/conf.py:22]: These won't work unless `docs/requirements.txt` include all the dev requirements (e.g. django). If I create a fresh virtual environment, install `docs/requirements.txt`, and then do `make docs`, I get this:

```
$ make docs
rm -rf docs/source
sphinx-apidoc -o docs/source/ mozilla_django_oidc
make -C docs clean
make[1]: Entering directory '/home/willkg/mozilla/mozilla-django-oidc/docs'
rm -rf _build/*
make[1]: Leaving directory '/home/willkg/mozilla/mozilla-django-oidc/docs'
make -C docs html
make[1]: Entering directory '/home/willkg/mozilla/mozilla-django-oidc/docs'
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v7.2.6

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/home/willkg/.pyenv/versions/3.10.12/lib/python3.10/site-packages/sphinx/config.py", line 358, in eval_config_file
    exec(code, namespace)  # NoQA: S102
  File "/home/willkg/mozilla/mozilla-django-oidc/docs/conf.py", line 20, in <module>
    import django  # noqa
ModuleNotFoundError: No module named 'django'

make[1]: *** [Makefile:53: html] Error 2
make[1]: Leaving directory '/home/willkg/mozilla/mozilla-django-oidc/docs'
make: *** [Makefile:38: docs] Error 2
```

**Review by willkg — CHANGES_REQUESTED** [review]
The docs don't build with these changes. If I install requirements from `requirements/requirements-dev.txt`, then they build. There are some warnings in the output.

File: docs/conf.py (hunk 2 above)
Comment (akatsoulas) [reply docs/conf.py:22]: Apologies for the back n forth here. I neglected to destroy my virtual environment which already had the packages. With the update `make docs` build without any errors for me. Hopefully one more pass :) ?

**Review by willkg — APPROVED** [review]
Looks great! No worries about the back-and-forth. :)

**Pull request merged**